sunpy/time/time.py

```diff
@@ -126,83 +129,71 @@ def _astropy_time(time):
     return time if isinstance(time, astropy.time.Time) else astropy.time.Time(parse_time(time))
 
 
-def _parse_dt64(dt):
-    """
-    Parse a single numpy datetime64 object
-    """
-    # Convert to microsecond precision because datetime cannot handle nanoseconds
-    return dt.astype('M8[us]').astype(datetime)
-
-
 @singledispatch
-def convert_time(time_string, **kwargs):
+def convert_time(time_string, format=None, **kwargs):
     # default case when no type matches
-    raise ValueError("'{tstr!s}' is not a valid time string!".format(tstr=time_string))
-
-
-@convert_time.register(pandas.Timestamp)
-def convert_time_pandasTimestamp(time_string, **kwargs):
-    return time_string.to_pydatetime()
+    try:
```

Comment:
Do we need this `try` block? can we not just delegate to astropy Time to raise an error if it does not understand the input?